source/src_pw/wf_igk.cpp

```diff
@@ -13,6 +13,9 @@ WF_igk::~WF_igk()
 		std::cout << " ~WF_igk()" << std::endl;
 	}
     delete[] g2kin;
+#ifdef __CUDA
+    cudaFree(this->d_g2kin);
+#endif
 }
```

Comment:
Be careful for using of d_g2kin, there is a memory leakage for no initial malloc.
Another question is can vector in STL be used in cuda or rocm?

Reply:
I have allocated memory for d_g2kin in wavefunc.cpp:L70, is it reasonable to do so? wavefunc inherits from WF_igk.

Reply:
Cuda does not yet support STL on the device side..

Reply:
Cuda does not yet support STL on the device side.

Reply:
> Cuda does not yet support STL on the device side..

Well it does have STL on both device and host, see https://github.com/NVIDIA/libcudacxx .
Personally, I have no experiece on how to use it.
FYI @dyzheng